sound/soc/sof/ipc4-pcm.c

```diff
@@ -59,19 +59,120 @@ int sof_ipc4_set_pipeline_state(struct snd_sof_dev *sdev, u32 id, u32 state)
 }
 EXPORT_SYMBOL(sof_ipc4_set_pipeline_state);
 
+static void
+sof_ipc4_add_pipeline_to_trigger_list(struct snd_sof_dev *sdev, int state,
+				      struct snd_sof_pipeline *spipe,
+				      struct ipc4_pipeline_set_state_data *trigger_list)
+{
+	struct snd_sof_widget *pipe_widget = spipe->pipe_widget;
+	struct sof_ipc4_pipeline *pipeline = pipe_widget->private;
+
+	if (pipeline->skip_during_fe_trigger)
+		return;
+
+	switch (state) {
+	case SOF_IPC4_PIPE_RUNNING:
+		/*
+		 * Trigger pipeline if all PCMs containing it are paused or if it is RUNNING
+		 * for the first time
+		 */
+		if (spipe->started_count == spipe->paused_count)
```

Comment:
I'm bad at this, but if the pipeline have on PCM user then if it should be started then the started_count should be 1 and the paused_count 0, or?
What if the spipe have, let's say 5 users, 1 is in started state and 2 in paused state, the other two does not matter as they are not running.
How this is working?

Reply:
I see now, this is done _before_ evaluating the started/paused count for this trigger, so it is done on the current before trigger counters.